decimal.go

```diff
@@ -565,7 +565,7 @@ func (d Decimal) QuoRem(d2 Decimal, precision int32) (Decimal, Decimal) {
 		panic("decimal division by 0")
 	}
 	scale := -precision
-	e := int64(d.exp - d2.exp - scale)
+	e := int64(d.exp) - int64(d2.exp) - int64(scale)
```

Comment:
Thanks for catching that! :3 LGTM!
Could you add a test case for this edge case? If you don't have time I can work on that.

Reply:
Sorry I do have no time on this. Could you please handle it?

Reply:
Sure will do, no problem :3